File: TsogosunProfileAdmin/MSPatronRewardsAdmin/Model/Dtos/ScratchCardDto.cs
using MSPatronRewardsAdmin.Shared;
using System;

namespace MSPatronRewardsAdmin.Model.Dtos
{
    public class ScratchCardDto : HasSiteId
    {
        public Guid ScratchCardUID { get; set; }
        public string Description { get; set; }
        public DateTime StartDateTime { get; set; }
        public DateTime EndDateTime { get; set; }
        public string ScratchCardImage { get; set; }
        public string ScratchCardTitle { get; set; }
        public string ScratchCardSubTitle { get; set; }
        public string TileImagePath { get; set; }
        public int ScratchCardTilesID { get; set; } = 1;
        public int NumberOfPrizes { get; set; } 
        public bool isActive { get; set; } = true;
        public string Status
        {
            get
            {
                return isActive == true ? "Active" : "Not Active";
            }
        }
    }
}

File: TsogosunProfileAdminWeb/src/app/model/Dtos/profile-rewards-admin/scratchCardDto.ts
import { HasSiteId } from "../../shared/hasSiteId";

export interface ScratchCardDto extends HasSiteId {
    scratchCardUID: string;
    description?: string | undefined;
    startDateTime: any;
    endDateTime: any;
    scratchCardImage?: string | undefined;
    scratchCardTitle?: string | undefined;
    scratchCardSubTitle?: string | undefined;
    tileImagePath?: string | undefined;
    numberOfPrizes: number;
    isActive: boolean;
    status?: string | undefined;
}

